Wire-iOS/Sources/UserInterface/MainController/ZClientViewController+FeatureChange.swift

```diff
@@ -21,15 +21,30 @@ import WireSyncEngine
 
 extension ZClientViewController {
     
-    func notifyUserOfDisabledAppLockIfNeeded() {
-        guard let appLock = ZMUserSession.shared()?.appLockController else {
-            return
+    var appLock: AppLockType? {
+        return _userSession?.appLockController
+    }
+    
+    var changeWarningViewController: AppLockChangeWarningViewController? {
+        guard let appLock = appLock else {
+            return nil
+        }
+        
+        return AppLockChangeWarningViewController(isAppLockActive: appLock.isActive) {
+            try? appLock.deletePasscode()
         }
-        if appLock.needsToNotifyUser && !appLock.isActive {
-            let warningVC = AppLockChangeWarningViewController(isAppLockActive: appLock.isActive)
-            warningVC.modalPresentationStyle = .fullScreen
-            present(warningVC, animated: false)
+    }
+    
+    func notifyUserOfDisabledAppLockIfNeeded() {
+        guard let appLock = appLock,
+            let warningVC = changeWarningViewController,
+            appLock.needsToNotifyUser,
+            !appLock.isActive else {
```

Comment:
To me it's nicer if we create the VC only after all the checks have been passed like this:

```suggestion
        guard 
            let appLock = appLock,
            appLock.needsToNotifyUser,
            !appLock.isActive
            let warningVC = changeWarningViewController, else {
```